spec/features/idv/in_person_threatmetrix_spec.rb

```diff
@@ -0,0 +1,305 @@
+        it_behaves_like 'initially shows the user the barcode page'
+
+        it_behaves_like 'shows the user the Please Call screen'
+
+        it 'does not allow the user to restart the IPP flow', allow_browser_log: true do
+          deactivate_profile_update_enrollment(status: :passed)
+
+          visit_idp_from_sp_with_ial2(sp)
+          expect(page).to have_current_path(idv_please_call_path)
+          page.visit('/verify/welcome')
+          expect(page).to have_current_path(idv_please_call_path)
+          page.visit('/verify/in_person/document_capture')
+          expect(page).to have_current_path(idv_please_call_path)
+        end
+
+        it 'shows the user the successful verification screen after passing TMX review',
+           allow_browser_log: true do
+          deactivate_profile_update_enrollment(status: :passed)
+          visit_idp_from_sp_with_ial2(sp)
+          expect(page).to have_current_path(idv_please_call_path)
+
+          review_pass.run(args: [user.uuid], config:)
+          page.visit('/verify/welcome')
+          expect(page).to have_current_path(idv_activated_path)
+        end
+      end
+
+      context 'User passes IPP and fails TMX Review' do
+        it_behaves_like 'initially shows the user the barcode page'
+
+        it_behaves_like 'shows the user the Please Call screen'
+
+        it 'does not allow the user to restart the flow', allow_browser_log: true do
+          deactivate_profile_update_enrollment(status: :passed)
+
+          # user revisits before fraud rejection
+          visit_idp_from_sp_with_ial2(sp)
+          expect(page).to have_current_path(idv_please_call_path)
+
+          # reject the user
+          review_reject.run(args: [user.uuid], config:)
+
+          # user revisits after fraud rejection
+          visit_idp_from_sp_with_ial2(sp)
+          expect(page).to have_current_path(idv_not_verified_path)
+        end
+      end
+
+      context 'User fails IPP and fails TMX review' do
+        it_behaves_like 'initially shows the user the barcode page'
+
+        it 'allows the user to restart the flow', allow_browser_log: true do
+          deactivate_profile_update_enrollment(status: :failed)
+
+          visit_idp_from_sp_with_ial2(sp)
+          expect(page).not_to have_current_path(idv_please_call_path)
+          # redo the flow:
+          complete_entire_ipp_flow(user, tmx_status)
+          expect(page).to have_current_path(idv_in_person_ready_to_verify_path)
+        end
+      end
+
+      context 'User cancels IPP after being deactivated for TMX review', allow_browser_log: true do
+        it 'allows the user to restart IPP' do
+          # set fraud review to pending:
+          profile.deactivate_for_fraud_review
+          profile.reload
+
+          # cancel the enrollment
+          click_link t('links.cancel')
+          # user can restart
+          click_on t('idv.cancel.actions.start_over')
+
+          expect(page).to have_current_path(idv_welcome_path)
```

Comment:
this now fails after the merge of the recent bug fix. The before action `confirm_fraud_pending` in the `please_call_controller` is not redirecting as `fraud_review_pending?` is true. (though that method would redirect us to the account page and not the welcome page)